rpc/types/src/other.rs

```diff
@@ -930,6 +952,178 @@ define_request_and_response! {
     }
 }
 
+//---------------------------------------------------------------------------------------------------- Request
+/// Other JSON requests.
+///
+/// This enum contains all [`crate::other`] requests.
+///
+/// See also: [`OtherResponse`].
+///
+/// # (De)serialization
+/// The `serde` implementation will (de)serialize from
+/// the inner variant itself, e.g. [`OtherRequest::SetLogLevel`]
+/// has the same (de)serialization as [`SetLogLevelRequest`].
+///
+/// ```rust
+/// use cuprate_rpc_types::other::*;
+///
+/// let request = OtherRequest::SetLogLevel(Default::default());
+/// let json = serde_json::to_string(&request).unwrap();
+/// assert_eq!(json, r#"{"level":0}"#);
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[cfg_attr(feature = "serde", serde(untagged))]
+#[allow(missing_docs)]
+pub enum OtherRequest {
+    GetHeight(GetHeightRequest),
+    GetTransactions(GetTransactionsRequest),
+    GetAltBlocksHashes(GetAltBlocksHashesRequest),
+    IsKeyImageSpent(IsKeyImageSpentRequest),
+    SendRawTransaction(SendRawTransactionRequest),
+    StartMining(StartMiningRequest),
+    StopMining(StopMiningRequest),
+    MiningStatus(MiningStatusRequest),
+    SaveBc(SaveBcRequest),
+    GetPeerList(GetPeerListRequest),
+    SetLogHashRate(SetLogHashRateRequest),
+    SetLogLevel(SetLogLevelRequest),
+    SetLogCategories(SetLogCategoriesRequest),
+    SetBootstrapDaemon(SetBootstrapDaemonRequest),
+    GetTransactionPool(GetTransactionPoolRequest),
+    GetTransactionPoolStats(GetTransactionPoolStatsRequest),
+    StopDaemon(StopDaemonRequest),
+    GetLimit(GetLimitRequest),
+    SetLimit(SetLimitRequest),
+    OutPeers(OutPeersRequest),
+    InPeers(InPeersRequest),
+    GetNetStats(GetNetStatsRequest),
+    GetOuts(GetOutsRequest),
+    Update(UpdateRequest),
+    PopBlocks(PopBlocksRequest),
+    GetTransactionPoolHashes(GetTransactionPoolHashesRequest),
+    GetPublicNodes(GetPublicNodesRequest),
+}
```

Comment:
Example of a request enum.

These request/response enums are used within `cuprate_rpc_interface`'s `tower::Service` API, not directly, but something like:
```rust
// inside `cuprate_rpc_interface`
enum RpcRequest {
    JsonRpc(cuprate_json_rpc::Request<JsonRpcRequest>),
    Other(OtherRequest),
    Binary(BinRequest),
}
```
then this broad `RpcRequest` type is used like such: `tower::Service<RpcRequest>`.